recipes/sdl/3.x/conanfile.py

```diff
@@ -0,0 +1,466 @@
+            tc.cache_variables["SDL_X11_XFIXES"] = self.options.xfixes
+            tc.cache_variables["SDL_X11_XRANDR"] = self.options.xrandr
+            tc.cache_variables["SDL_X11_XSCRNSAVER"] = self.options.xscrnsaver
+            tc.cache_variables["SDL_X11_XSHAPE"] = self.options.xshape
+            tc.cache_variables["SDL_X11_XSYNC"] = self.options.xsync
+
+        with_wayland = self.options.get_safe("wayland", False)
+        tc.cache_variables["SDL_WAYLAND"] = with_wayland
+        if with_wayland:
+            tc.cache_variables["SDL_WAYLAND_SHARED"] = self.dependencies["wayland"].options.shared
+        if not with_x11 and not with_wayland:
+            # Disable windowing support:
+            # https://github.com/libsdl-org/SDL/blob/main/docs/README-cmake.md#cmake-fails-to-build-without-x11-or-wayland-support
+            tc.cache_variables["SDL_UNIX_CONSOLE_BUILD"] = True
+
+        tc.generate()
+        deps = CMakeDeps(self)
+        deps.set_property("libusb", "cmake_target_name", "LibUSB::LibUSB")
+        deps.set_property("libusb", "cmake_file_name", "LibUSB")
+        deps.generate()
+        pcdeps = PkgConfigDeps(self)
+        pcdeps.generate()
+
+    def _patch_sources(self):
+        # TODO: Once new CMakeDeps is default, remove this
+        # Right now this would fail at runtime for shared libusb, but we're
+        # validating it out for now. The new incubating cmakedeps in 2.12 is required
+        # to build that configuration, else remove the failing line in the old one
+        if conan_version < "2.12" or not self.conf.get("tools.cmake.cmakedeps:new"):
+            replace_in_file(self, os.path.join(self.source_folder, "cmake", "sdlchecks.cmake"),
+                        "target_get_dynamic_library(dynamic_libusb LibUSB::LibUSB)",
+                        "")
+
+    def build(self):
+        self._patch_sources()
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        copy(self, "LICENSE.txt", self.source_folder, os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
+        cmake.install()
+
+    @property
+    def _is_clang_cl(self):
+        return self.settings.os == "Windows" and self.settings.compiler == "clang" and \
+            self.settings.compiler.get_safe("runtime")
+
+    def package_info(self):
+        sdl_lib_name = "SDL3"
+        if (is_msvc(self) or self._is_clang_cl) and not self.options.shared:
+            sdl_lib_name = f"{sdl_lib_name}-static"
+        self.cpp_info.libs = [sdl_lib_name]
+        self.cpp_info.set_property("cmake_file_name", "SDL3")
+        self.cpp_info.set_property("cmake_target_name", "SDL3::SDL3")
+        self.cpp_info.set_property("cmake_target_aliases", [
+            "SDL3::SDL3-shared" if self.options.shared else "SDL3::SDL3-static",
+            # Modelling the headers as a target is not quite feasible, so use an alias instead
+            "SDL3::Headers"
```

Comment:
Could be done with a component that sets `includedirs` and no `libs` or `requires`, no?

Reply:
Yes, that's another approach and I might just go for it after we check that the rest of the recipe looks ok - having only the global component would be better but alas

Reply:
This will cause an issue if you ever add sdl2-compat to cci

Reply:
Looks like I was wrong, adding it as a proper component was not quite as involved as I first thought!